src/celengine/textlayout.cpp

```diff
@@ -0,0 +1,258 @@
+// textlayout.cpp
+//
+// Copyright (C) 20123-present, the Celestia Development Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+#include <vector>
+
+#include "textlayout.h"
+#include "celutil/utf8.h"
+
+namespace celestia::engine
+{
+
+TextLayout::TextLayout(int screenDpi, HorizontalAlignment halign) : screenDpi(static_cast<float>(screenDpi)), horizontalAlignment(halign), positionX(0.0f), positionY(0.0f), alignmentEdgeX(0.0f)
```

Comment:
```
TextLayout::TextLayout(int screenDpi, HorizontalAlignment halign) :
    screenDpi(static_cast<float>(screenDpi)),
    horizontalAlignment(halign)
{
}
```
The following should be done in .h file: `positionX(0.0f), positionY(0.0f), alignmentEdgeX(0.0f)`